### PR TITLE
[Serverless] Improve the adaptive flush strategy of the serverless extension

### PR DESCRIPTION
**Pull request by jcstorms1**
### What does this PR do?

Increases the amount of invocations required from for the extension to transition into a periodic flush strategy.

- Increased the amount of invocation time stamps stored from 20 to 30.
- Increased periodic flush threshold from 3 to 20 invocations.

### Motivation

Hand testing a lambda function was transitioning the extension into a periodic flush strategy too quickly since users would often times invoke the function more than 3 times.  This could lead to confusion since not all invocations would appear on the serverless page until reaching the flush interval of 20 seconds.

### Describe how to test/QA your changes

Tested by manually deploying the extension and hand testing.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by nhinsch — DISMISSED** [review]

File: pkg/serverless/daemon/interval_test.go [hunk 1]
```diff
@@ -87,15 +93,15 @@ func TestInvocationInterval(t *testing.T) {
...
+	for i := 0; i < 19; i++ {
 		time.Sleep(100 * time.Millisecond)
 		d.lastInvocations = append(d.lastInvocations, time.Now())
 		assert.Equal(time.Duration(0), d.InvocationInterval(), "we should not compute any interval just yet since we don't have enough data")
 	}
 	time.Sleep(100 * time.Millisecond)
 	d.lastInvocations = append(d.lastInvocations, time.Now())
 
-	//	assert.Equal(d.InvocationInterval(), time.Duration(0), "we should not compute any interval just yet since we don't have enough data")
+	//  assert.Equal(d.InvocationInterval(), time.Duration(0), "we should not compute any interval just yet since we don't have enough data")
```
Comment (nhinsch) [inline pkg/serverless/daemon/interval_test.go:104]: What is this comment?

Comment (jcstorms1) [reply pkg/serverless/daemon/interval_test.go:104]: I don't know. I guess it can be removed.

File: pkg/serverless/daemon/interval_test.go [hunk 2]
```diff
@@ -23,22 +23,28 @@ func TestAutoSelectStrategy(t *testing.T) {
...
 	assert.Equal((&flush.AtTheEnd{}).String(), d.AutoSelectStrategy().String(), "not the good strategy has been selected") // default strategy
 
-	assert.True(d.StoreInvocationTime(now.Add(-time.Second * 140)))
+	assert.True(d.StoreInvocationTime(now.Add(time.Second * 18)))
 	assert.Equal((&flush.AtTheEnd{}).String(), d.AutoSelectStrategy().String(), "not the good strategy has been selected")
-	assert.True(d.StoreInvocationTime(now.Add(-time.Second * 70)))
+	assert.True(d.StoreInvocationTime(now.Add(time.Second * 19)))
 	assert.Equal((&flush.AtTheEnd{}).String(), d.AutoSelectStrategy().String(), "not the good strategy has been selected")
 
 	// add a third invocation, after this, we have enough data to decide to switch
```
Comment (nhinsch) [inline pkg/serverless/daemon/interval_test.go:42]: Looks like this comment is out of date now, probably best to just remove it and let the code speak for itself.

File: pkg/serverless/daemon/interval_test.go [hunk 3]
```diff
@@ -87,15 +93,14 @@ func TestInvocationInterval(t *testing.T) {
 	// first scenario, validate that we're not computing the interval if we only have 2 invocations done
```
Comment (nhinsch) [inline pkg/serverless/daemon/interval_test.go:93]: Likewise this comment is also wrong now

**Review by nhinsch — APPROVED** [review]

**Pull request merged**